### DISCUSSION TITLE
Command's listener won't know it's cancelled when cancelling a command.

### DISCUSSION THREAD

**Discussion by shrimptttttttttt** [redis/lettuce#2471] [General]
I'm a 3.5.0.Final version lettuce user, Here is my question.

I dispatch a Command which has a listener, during some cases, I need to cancel the command. I expect the command's listener will know it's cancelled when it's notified. Tests show that, listener won't get the cancel status.

I think the problem is the Command extends AbstractFuture, Command doesn't override AbstractFuture's cancel method, but Command has implemented it's own 'isCancelled', which is separated. 

It's high risky for me to upgrade lettuce version, so can you please help me calarify how should I fix this? Thanks a lot!!!

**Comment by shrimptttttttttt** [comment]
Any body can have a look at this? @wg

**Reply by mp911de** [reply]
Version 3.5 is not maintained for several years anymore. Please upgrade to version 6.2.x at your earliest convenience.
